app/models/user_role.rb

```diff
@@ -1,10 +1,17 @@
 class UserRole < ActiveRecord::Base
+  has_paper_trail class_name: 'Version'
+
   belongs_to :user
   belongs_to :role
 
   validates :role_id, uniqueness: { scope: :user_id }
+  before_save :set_expires_at_time, if: lambda { expires_at.present? }
 
   def self.create_user_role(user_id, role_id)
     create(user_id: user_id, role_id: role_id)
   end
+
+  def set_expires_at_time
+    self.expires_at = self.expires_at.change(hour: 20)
+  end
```

Comment:
We allow datetime (instead of date) for flexibility, enforcing `8pm` on all expiry dates defeats that purpose. Rather the front-end should send the requested day and hour when setting the expiry.

On the first go, the front-end will probably only allow selecting dates and default to `8pm`. But in the future we may want to add more granular controls on the UI. This would bypass anything we set from there.